### PR TITLE
Pin typos version to latest currently available tag v1.24.5

### LINKED ISSUE #1485: Pin version used for CI spellchecking to avoid surprise failures after updates
> We could pin between releases, and on each release cycle update the version and fix any flagged typos. Might be a "best of both worlds" approach so PRs don't fail when `typos` pushes a new version.

_Originally posted by @bpaul4 in https://github.com/IDAES/idaes-pse/issues/1482#issuecomment-2329555598_

Personally I think @bpaul4's suggestion is very sensible, and I suggest we adopt this to avoid having to deal with the concrete possibility of unrelated PRs starting to fail whenever a new version of the `typos` tool we use for spellchecking in CI is released

### PR DESCRIPTION
**Pull request by lbianchi-lbl**
### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
@bpaul4 I've added you and myself as code owners for the typos config file, so that we're added as reviewers whenever it's modified.

**Review by bpaul4 — APPROVED** [review]
This is great, thank you @lbianchi-lbl for opening the PR. Noting that we will probably reverse the "workpace" exception once #1484 is merged.

**Comment by lbianchi-lbl** [comment]
> This is great, thank you @lbianchi-lbl for opening the PR. Noting that we will probably reverse the "workpace" exception once #1484 is merged.

Yeah, I was planning on removing the DMF exceptions directly within #1484 as a further double-check that all affected files have been removed.

**Pull request merged**
